docs/examples/tutorial/profiling_tutorial/calc_pi_3.py

```diff
@@ -0,0 +1,14 @@
+# cython: profile=True
+
+@cython.cfunc
+@cython.inline
+@cython.exceptval(-1.0)
+def recip_square(i: cython.int) -> cython.double:
+    return 1. / (cython.cast(cython.long, i) * i)
```

Comment:
I have found out that example currently present in master branch is failing with following error:
```
Traceback (most recent call last):
  File "/home/matus/dev/cython/docs/examples/tutorial/profiling_tutorial/test/profile_2.py", line 8, in <module>
    cProfile.runctx("calc_pi.approx_pi()", globals(), locals(), "Profile.prof")
  File "/usr/lib/python3.9/cProfile.py", line 19, in runctx
    return _pyprofile._Utils(Profile).runctx(statement, globals, locals,
  File "/usr/lib/python3.9/profile.py", line 62, in runctx
    prof.runctx(statement, globals, locals)
  File "/usr/lib/python3.9/cProfile.py", line 100, in runctx
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "calc_pi.pyx", line 11, in calc_pi.approx_pi
    val += recip_square(k)
  File "calc_pi.pyx", line 5, in calc_pi.recip_square
    return 1. / (i * i)
ZeroDivisionError: float division
```

It seems that it is due overflow of `i * i` expression. Casting to `long` fixes the issue.

Reply:
1) can we cast whole `i*i` statement? it'll look better
2) then why not to use `long` instead of `int`?
3) can't we prevent passing such values that cause overflow? We are controlling them in `approx_pi` AFAIU

Reply:
> can we cast whole i*i statement? it'll look better
   
`cython.cast(cython.long, i * i))` does not work because it is yelding `(long) (i * i)` C code which is still crashing.

> then why not to use long instead of int?

I was thinking about that. For now I decided to leave example as close to original as possible (with `int` parameters of function). I leave on @scoder what is better

> can't we prevent passing such values that cause overflow? We are controlling them in approx_pi AFAIU

We can but the simple example is not doing that. I think the best is to keep it simple and just change type to long or cast it.